src/app/integrations/edit-page/step-configure/step-configure.component.ts

```diff
@@ -126,6 +128,24 @@ export class IntegrationsStepConfigureComponent extends FlowPage
     });
   }
 
+  fetchDataShapesFor(step: Step) {
+    return this.integrationSupport.fetchMetadata(
+      step.connection,
+      step.action,
+      step.configuredProperties || {},
+    )
+    .toPromise()
+    .then(response => {
```

Comment:
We try to encapsulate the HTTP call in the service but return the response here. It would be awesome if we just returned the value.